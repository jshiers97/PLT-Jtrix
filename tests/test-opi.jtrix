void main() {
	Matrix<int> a;
	Matrix<int> b;
	Matrix<int> sum;
	Matrix<int> diff;
	int i;
	int j;

	a = [[1, 2];[3, 4]];
	b = [[5, 6];[7, 8]];

	sum = a + b;
	diff = a - b;

	println("Sum");
	for(i = 0; i < 2; i = i++) {
		for(j = 0; j < 2; j = j++) {
			print(sum[i][j]);	
		}
	}

	println("Diff");
	for(i = 0; i < 2; i = i++) {
		for(j = 0; j < 2; j = j++) {
			print(diff[i][j]);	
		}
	}

	free(sum);
	free(diff);
	free(a);
	free(b);

}
