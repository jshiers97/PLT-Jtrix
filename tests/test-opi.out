Sum
6
8
10
12
Diff
-4
-4
-4
-4
